### DISCUSSION TITLE
Penilaian Capstone Project Machine Learning

### DISCUSSION THREAD

**Discussion by JonathanCP123** [teamalgoritma/community#909] [Q&A] [answered]
Halo Kak, saya baru baca ulang evaluasi saya di capstone project machine learning dan saya baru sadar bahwa dikatakan 
"...pada saat melakukan evaluasi neural network, metrics hanya dihitung pada training dataset, harap bisa melengkapi juga dengan perhitungan metrics pada validation set."

evaluasi neural network saya hitung menggunakan validation set saya sendiri
bagaimana caranya saya bisa melakukan evaluasi neural network saya menggunakan data test kalao di data test tidak ada data strength untuk di compare ?

**Comment by saskia-dwi-ulfah** [comment, marked as answer]
Dear @JonathanCP123 

Terima kasih sudah bertanya melalui Github Community Algoritma. 
Terkait feedback capstone machine learning Bapak berikut beberapa poin tambahan yang dapat kami sampaikan:

- Dari link Rpubs yang Bapak lampirkan, Bapak sudah menggunakan fungsi `evaluate()` untuk  membandingkan metrics data training dan data validation.
![image](https://github.com/user-attachments/assets/f4fef696-5eff-4c9e-8548-9fc57439987e)
- Selanjutnya, Bapak menggunakan fungsi `postResample()` dari `caret` untuk menghitung metrics yang lebih terperinci (RMSE, Rsquared, MAE) untuk mengevaluasi prediksi model pada data training.
![image](https://github.com/user-attachments/assets/7ac2903f-90d8-4056-ac52-53e0acb1ae99)

- Terkait feedback, _"...pada saat melakukan evaluasi neural network, metrics hanya dihitung pada training dataset, harap bisa melengkapi juga dengan perhitungan metrics pada validation set."_, harapannya Bapak juga bisa menambahkan fungsi `postResample()` untuk evaluasi prediksi model pada data validation Bapak, yang dalam hal ini disimpan pada variabel `keras_test`. Harapannya, dari hasil evaluasi neural network ini Bapak bisa memenuhi rubrics  
   * [0/3] MAE in (your own) validation dataset reach < 4.
   * [0/3] R-squared in (your own) validation dataset reach > 90%.

Demikian yang dapat kami sampaikan.

Best regards,
Team Algoritma
